CommonMC/src/SelectRecoMC_module.cc

```diff
@@ -278,7 +286,14 @@ namespace mu2e {
 
   void SelectRecoMC::fillTSHMC(TrkStrawHitMC& tshmc, size_t isdmc, size_t isp,StrawDigiMC const& sdmc,
       Tracker const& tracker, std::shared_ptr<const StrawResponse>const& srep ) {
+    // always record the index, to avoid downstream corruptions
     tshmc._sdmcindex = isdmc;
+    // propagate interpretability of StrawDigiMC to TrkStrawHitMC
+    tshmc._provenance = static_cast<TrkStrawHitProvenance>(sdmc.provenance(), true);
```

Comment:
This relies on two independent enums having the same enumerator values meaning the same thing.  It works now and it works so long as we never change one enum without making the matching changing the other.  While I think that the odds are good that we would get a way with this, it just feels dangerous.

At line 80 of KalSeedMC.hh, try the following;
using TrkStrawHitProvenance = StrawDigiProvenance;

Then get rid of all traces of TrackStrawHitProvenenance and add a comment to StrawDigiMC alerting future maintainers that this using is present and they should be aware of it.

If it works this still feels kind of icky because the name of the detail class is now a lie but it's far less icky than where we are now.

Reply:
See out-of-thread comment.